File: src/main/java/com/vsharstuk/pizzatoppings/mapper/ToppingMapper.java
package com.vsharstuk.pizzatoppings.mapper;

import com.vsharstuk.pizzatoppings.dto.ToppingDto;
import com.vsharstuk.pizzatoppings.entity.Topping;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.ReportingPolicy;

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.ERROR)
public interface ToppingMapper {

    @Mapping(target = "id", source = "id")
    @Mapping(target = "name", source = "name")
    ToppingDto toToppingDto(Topping entity);


    @Mapping(target = "id", source = "id")
    @Mapping(target = "name", source = "name")
    @Mapping(target = "users", ignore = true)
    Topping toToppingEntity(ToppingDto dto);
}

File: src/main/kotlin/com/vsharstuk/pizzatoppings/mapper/ToppingMapper.kt
package com.vsharstuk.pizzatoppings.mapper

import com.vsharstuk.pizzatoppings.dto.ToppingDto
import com.vsharstuk.pizzatoppings.entity.Topping
import org.springframework.stereotype.Component


@Component
open class ToppingMapper {

    fun toToppingDto(entity: Topping?): ToppingDto?{
        val dto = ToppingDto()
        dto.id = entity?.id
        dto.name = entity?.name
        return dto
    }

    fun toToppingEntity(dto: ToppingDto?): Topping?{
        val entity = Topping()
        entity.id = dto?.id
        entity.name = dto?.name
        return entity
    }
}
